examples/stdlib/stream/fastexp.effekt

```diff
@@ -0,0 +1,95 @@
+import stream
+
+/// Least significant bits
+def eachLSB(n: Int): Unit / emit[Bool] = {
+  if (n <= 0) {
+    do emit(false)
+  } else {
+    var tmp = n
+    while (tmp > 0) {
+      do emit(if (tmp.mod(2) == 1) true else false)
+      tmp = tmp / 2
+    }
+  }
+}
+
+/// Most significant bits
+def eachMSB(n: Int): Unit / emit[Bool] = {
+  if (n <= 0) {
+    do emit(false)
+  } else {
+    // First, calculate the highest bit position
+    var highest = 0
+    var m = n
+    while (m > 0) {
+      highest = highest + 1
+      m = m / 2
+    }
+
+    // Then emit from MSB to LSB
+    for[Int] { range(0, highest) } { i =>
+      val index = highest - i
+      val bit = n.bitwiseShr(index - 1).mod(2) == 1
+      do emit(bit)
+    }
+  }
+}
+
+def fastexp(n: Int, k: Int) = product {
+  stream::zip[Int, Bool] {n.iterate { x => x * x }} {k.eachLSB} {
+    case res, true => do emit(res)
+    case res, false => ()
+  }
+}
+
+def main() = {
+  /// Computes n^exp for integers. (from #923!)
+  def pow(n: Int, exp: Int): Int = {
+    def go(n: Int, exp: Int, acc: Int): Int = {
+      if (exp == 0) {
+        acc
+      } else if (mod(exp, 2) == 0) {
+        go(n * n, exp / 2, acc)
+      } else {
+        go(n * n, exp / 2, acc * n)
+      }
+    }
+    go(n, exp, 1)
+  }
```

Comment:
Taken verbatim from #923, I really ought to rebase :)

By the way, as an exercise in algorithms, notice that the `pow` here and the `fastexp` above are completely isomorphic, always doing the exact same steps :)